launcher/Version.cpp

```diff
@@ -74,12 +74,28 @@ bool Version::operator!=(const Version &other) const
 void Version::parse()
 {
     m_sections.clear();
-
-    // FIXME: this is bad. versions can contain a lot more separators...
-    QStringList parts = m_string.split('.');
-
-    for (const auto& part : parts)
-    {
-        m_sections.append(Section(part));
+    QString currentSection;
+    auto classChange = [] (QChar lastChar, QChar currentChar) {
+        return (( lastChar.isLetter() && currentChar.isDigit() ) || (lastChar.isDigit() && currentChar.isLetter()) );
+    };
+    for (int i = 0; i < m_string.size(); ++i) {
+        if(m_string[i].isDigit() || m_string[i].isLetter()){
```

Comment:
you can substitute all the `m_string[i]` for a line at the beginning of the loop with `const auto& current_char = m_string.at(i);`

Reply:
done in https://github.com/PrismLauncher/PrismLauncher/pull/751/commits/5d558702706e9ad464340879ec8a67c73f1892a9